packages/backend/src/videos/videos.service.ts

```diff
@@ -1,80 +1,155 @@
+		}
+
+		const videos = query.getMany();
+		return videos;
+	}
+
+	async getAllCategoryList() {
+		return await getRepository(CategoryEntity)
+			.createQueryBuilder("category")
+			.leftJoinAndSelect("category.videos", "videos")
+			.getMany();
+	}
+
+	async getStream(id: string, res: any, req: any) {
+		const video = await this.getSingleVideo(id);
+		const path = 'uploads/video/p720-angular-cli-dla-programistow-java-angular-w-45-min.mp4';
+		const stat = fs.statSync(path);
+		const range = req.headers.range;
+		const fileSize = stat.size;
+		try {
+			if (range) {
+				const parts = range.replace(/bytes=/, '').split('-');
+				const start = parseInt(parts[0], 10);
+				const end = parts[1] ? parseInt(parts[1], 10) : fileSize - 1;
+				const chunksize = end - start + 1;
+				const VideoReadStream = fs.createReadStream(path, { start, end });
+				const head = {
+					'Content-Range': `bytes ${start}-${end}/${fileSize}`,
+					'Accept-Ranges': 'bytes',
+					'Content-Length': chunksize,
+					'Content-Type': 'video/mp4',
+				};
+				res.writeHead(206, head);
+				VideoReadStream.pipe(res);
+			}
+			else {
+				const head = {
+					'Content-length': fileSize,
+					'Content-Type': 'video/mp4',
+				};
+				res.writeHead(200, head);
+				fs.createReadStream(path).pipe(res);
+			}
+		} catch (err) {
+			return err;
+		}
+	}
+
+	async getSingleVideo(id: string): Promise<VideosEntity> {
+		return await this.videosRepository.findOne(id);
+	}
+
+	async addNewVideoFromFiles(): Promise<VideosEntity[]> {
+		try {
+			if (!fs.existsSync(defaultVideosFolder))
+				return;
+
+			const files = fs.readdirSync(defaultVideosFolder).filter(file => file.endsWith(".mp4"));
+			if (!files.length)
+				return;
+
+			const videos = [];
+
+			for (let i = 0; i < files.length; i++) {
+				const file = files[i];
+
+				const videoEntity = await this.videosRepository.findOne({
+					where: [
+						{ urlVideo: file },
+					]
+				});
+
+				if (videoEntity)
+					continue;
+
+				let video = new VideosEntity();
+
+				const fileUrl = defaultVideosFolder + file;
+				const photoUrl = defaultPhotoFolder + file.replace(".mp4", ".jpeg");
+				const statFile = fs.statSync(fileUrl);
+
+				video.title = file.substring(0, file.lastIndexOf("."));
+				video.urlVideo = file;
+				video.urlTrailer = "";
+				video.urlPhoto = fs.existsSync(photoUrl) ? photoUrl : "";
+				video.dateCreation = statFile.birthtime.toLocaleDateString().replace(".", "-").replace(".", "-");
+				video.duration = await VideosService.getDurationFromFiles(fileUrl);
+				video.description = "";
+				video.country = "pl";
+				video.language = "pl";
+
+				videos.push(await this.videosRepository.save(video));
+
+			}
+			return videos;
+		} catch (error) {
+			throw new HttpException(error, HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+
+
+	}
+
+	private static async getDurationFromFiles(path: string): Promise<number> {
+		const fs = require("fs").promises;

```

Comment:
co oznacza fs? w sensie od czego to jest skrót ?

Reply:
fs chyba filesystem ale to nazwa biblioteki